Swift 4.2

douglashill opened the pull request:
This migrates all three targets to Swift 4.2. It also removes a deprecated API call in the example app that was not needed and resulted in a warning with Xcode 10 — see https://developer.apple.com/documentation/uikit/uiapplication/1622923-setstatusbarstyle

douglashill committed:
Migrate to Swift 4.2

douglashill committed:
Stop using deprecated API

UIApplication statusBarStyle is deprecated, and didn’t do anything anyway unless UIViewControllerBasedStatusBarAppearance was set to NO in the Info.plist (and it wasn’t in this example).

UINavigationController returns a preferredStatusBarStyle based on its navigationBar’s barStyle.

Boerworz reviewed (changes requested):
Thank you for doing this, Douglas! 🌟 It's been on my todo list for a while 😅

Unfortunately, the changes in this pull request break Objective-C compatibility due to the limited `@objc` inference in Swift 4. It seems like the fix is to add `@objcMembers` to the public classes in the `GagatObjectiveC` target. More information about the `@objc`-related changes in Swift 4 can be found in [the original proposal](
https://github.com/apple/swift-evolution/blob/master/proposals/0160-objc-inference.md) (where the [Re-enabling `@objc` inference within a class hierarchy](https://github.com/apple/swift-evolution/blob/master/proposals/0160-objc-inference.md#re-enabling-objc-inference-within-a-class-hierarchy) section is especially relevant for this case).

If you could make the needed changes I'd be happy to verify and get this merged!

douglashill committed:
Add Objective-C example

This makes is easier to know if changes break the Objective-C framework.

This example is simpler than the Swift one.

douglashill committed:
Let Xcode add IDEDidComputeMac32BitWarning

douglashill committed:
Expose public API to Objective-C

Fixes Objective-C framework API not being accessible after switching to Swift 4.2.

douglashill commented:
Since the public API is so small I thought it made more sense to add `@objc` to each public method rather than `@objcMembers` on the classes.

I also added an Objective-C example so it’s easier to test if the Objective-C framework is working. If you don’t want that in the repo I’m happy to delete that commit, but I think it’s worth the noise.

Boerworz commented:
Thanks again for taking the time to make these changes, Douglas! Adding an Objective-C example was a great idea 👌

I've tested the changes locally and everything seems to work as expected. I'll get a new release out soon 🚀

douglashill commented:
Yeah if the Objective-C wrapper is being maintained there needs to be an easy way to check it works.

Thank you for creating and open-sourcing Gagat!
